scipy/sparse/construct.py

```diff
@@ -622,7 +622,7 @@ def bmat(blocks, format=None, dtype=None):
     return coo_matrix((data, (row, col)), shape=shape).asformat(format)
 
 
-def block_diag(mats, format=None, dtype=None):
+def block_diag(mats, format='coo', dtype=None):
```

Comment:
We shouldn't change the default argument for `format` here. By convention, `format=None` means "I don't care what format you return", so it shouldn't affect the rest of your PR.